src/Symfony/Component/Serializer/Encoder/ChainDecoder.php

```diff
@@ -32,6 +32,8 @@ public function __construct(array $decoders = array())
 
     /**
      * {@inheritdoc}
+     * 
+     * @throws RuntimeException If no decoder is found.
```

Comment:
> I don't agree with that: you may handle differently an exception coming from a serialization error than a generic one.

`@throws Exception` doesn't help here anyway. If you handle specific exception, you know internals of Serializer.